frontend/app/src/pages/main/tenant-settings/webhooks/index.tsx

```diff
@@ -0,0 +1,205 @@
+  const listWebhookWorkersQuery = useQuery({
+    ...queries.webhookWorkers.list(tenant.metadata.id),
+  });
+
+  return (
+    <div className="flex-grow h-full w-full">
+      <div className="mx-auto max-w-7xl py-8 px-4 sm:px-6 lg:px-8">
+        <div className="flex flex-row justify-between items-center">
+          <h2 className="text-2xl font-semibold leading-tight text-foreground">
+            Webhooks
+          </h2>
+
+          <Button
+            key="create-webhook-worker"
+            onClick={() => setShowCreateTokenDialog(true)}
+          >
+            Create Webhook Endpoint
+          </Button>
+        </div>
+        <p className="text-gray-700 dark:text-gray-300 my-4">
+          Assign webhook workers to workflows.
+        </p>
+        <Separator className="my-4" />
+
+        <div className="grid gap-2 grid-cols-1 sm:grid-cols-2">
+          <div className="">
+            {listWebhookWorkersQuery.isLoading && 'Loading...'}
+          </div>
+          <div className="">{listWebhookWorkersQuery.isError && 'Error'}</div>
+          {listWebhookWorkersQuery.data?.rows?.map((worker) => (
+            <div key={worker.metadata!.id}>
+              <div className="flex flex-row justify-between items-center">
+                <Card>
+                  <CardHeader>
+                    <CardTitle className="flex items-center justify-between">
+                      {worker.name}
+                      <DropdownMenu>
+                        <DropdownMenuTrigger>
+                          <Button
+                            aria-label="Workflow Actions"
+                            size="icon"
+                            variant="ghost"
+                          >
+                            <BiDotsVertical />
+                          </Button>
+                        </DropdownMenuTrigger>
+                        <DropdownMenuContent>
+                          <DropdownMenuItem
+                            onClick={() => {
+                              setShowDeleteTokenDialog(worker.metadata.id);
+                            }}
+                          >
+                            Delete
+                          </DropdownMenuItem>
+                        </DropdownMenuContent>
+                      </DropdownMenu>
+                    </CardTitle>
+                    <CardDescription>
+                      <div className="text-sm mt-2 font-mono">
+                        {worker.metadata.id}
+                      </div>
+                      <div className="text-sm mt-2">{worker.url}</div>
+                    </CardDescription>
+                  </CardHeader>
+                </Card>
+              </div>
+            </div>
+          ))}
+
+          <CreateWebhookWorker
+            tenant={tenant.metadata.id}
+            showDialog={showCreateTokenDialog}
+            setShowDialog={setShowCreateTokenDialog}
+            onSuccess={() => {
+              void listWebhookWorkersQuery.refetch();
```

Comment:
nit: invalidate the query instead of refetching

Reply:
I do need to refetch the data so that the newly added item shows, no? Or what am I missing?

Reply:
```
queryClient.invalidateQueries({
        queryKey: queries.workflowRuns.get(
          tenant.metadata.id,
          workflowRun.metadata.id,
        ).queryKey,
      });

```

Something like this will invalidate all requests with that query and refetch accordingly, i think refetch will just refetch that specific request

Reply:
minor thing and not required for this pr

Reply:
if the new response is cached it would not matter though I think, is it not cached atm?